src/System.Runtime.Extensions/tests/System/Diagnostics/Stopwatch.cs

```diff
@@ -69,9 +69,17 @@ public static void StartNewAndRestart()
             TimeSpan elapsedSinceStart = watch.Elapsed;
             Assert.True(elapsedSinceStart > TimeSpan.Zero);
 
-            watch.Restart();
-            Assert.True(watch.IsRunning);
-            Assert.True(watch.Elapsed < elapsedSinceStart);
+            for (int attempt = 0; attempt < 5; attempt++)  // The comparison below could fail if we get very unlucky with when the thread gets preempted
+            {
+                watch.Restart();
+                Assert.True(watch.IsRunning);
+                try
+                {
+                    Assert.True(watch.Elapsed < elapsedSinceStart);
```

Comment:
Does this test anything? If it fails on all 5 attempts we will just catch it and move along, no?